src/msg/async/ProtocolV1.cc

```diff
@@ -1174,6 +1177,7 @@ ssize_t ProtocolV1::write_message(Message *m, bufferlist &bl, bool more) {
 }
 
 void ProtocolV1::requeue_sent() {
+  write_in_progress = false;
```

Comment:
i don't think this line is necessary. it has two callers

- ProtocolV1::replace(): `exproto->write_in_progress` is already reset before the lambda is scheduled.
- ProtocolV1::fault(): `write_in_progress` will always be reset before we leave `ProtocolV1::write_event()`.

and because `conn->handle_write()` will always be scheduled after `write_in_progress` is set. can we assume that we don't need to reset this flag in the fault handling paths?

Reply:
I also had the same doubt but I think we can't assume that `conn->handle_write()` will always be called after setting `write_in_progress = true`. Imagine a situation where a read event is executed after `send_message` and before the write event and detects a connection failure.